main.go

```diff
@@ -83,3 +117,19 @@ func VerboseLog(msg string) {
 		fmt.Println("*", msg)
 	}
 }
+
+func WriteStringProperty(b *bytes.Buffer, name string, value string) {
+	if value != "" {
+		WriteProperty(b, name, value)
+	}
+}
+
+func WriteUInt8Property(b *bytes.Buffer, name string, value uint8) {
+	if value != 0 {
```

Comment:
Well, I see your point. I think the headers are valuable, and I think they belong in the same file, and therefore (following your suggestion), I think JSON is the better format.